core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -527,6 +517,10 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
         }
     }
 
+    public void addOnStartedListener(Runnable runnable) {
+        onStartedListeners.add(runnable);
```

Comment:
our plugin API on the node level is pull based not push. Can we add a method to pull these from `ClusterPlugin.java` that way we are consistent and we fully control who calls this method.